lib/rig_api/v1/responses.ex

```diff
@@ -36,18 +35,16 @@ defmodule RigApi.V1.Responses do
 
   Note that body has to contain following field `"rig": { "correlation": "_id_" }`.
   """
-  def create(conn, message) do
-    with {:ok, cloud_event} <- CloudEvent.parse(message),
-         {:ok, rig_metadata} <- Map.fetch(message, "rig"),
-         {:ok, correlation_id} <- Map.fetch(rig_metadata, "correlation"),
-         {:ok, deserialized_pid} <- Codec.deserialize(correlation_id) do
-      Logger.debug(fn ->
-        "HTTP response via internal HTTP to #{inspect(deserialized_pid)}: #{inspect(message)}"
-      end)
-
-      send(deserialized_pid, {:response_received, cloud_event.json})
-      send_resp(conn, :accepted, "message sent to correlated reverse proxy request")
-    else
+  def create(%{req_headers: req_headers} = conn, message) do
+    case ResponseFromParser.parse(req_headers, message) do
+      {deserialized_pid, response_code, response} ->
+        Logger.debug(fn ->
+          "HTTP response via internal HTTP to #{inspect(deserialized_pid)}: #{inspect(message)}"
+        end)
+
+        send(deserialized_pid, {:response_received, response, response_code})
+        send_resp(conn, :accepted, "message sent to correlated reverse proxy request")
+
```

Comment:
I didn't notice that before, but I guess this is not how versions are supposed to work :D I'd not touch v1 and v2 and instead create a v3, since this is clearly a breaking change. With 3.0 we probably no longer care about v1 and v2, so you can remove them in this case, but I'd still call the new behavior v3

Reply:
Okay, but I would keep v1 and v2 and remove them in separate issue as there is more deprecated things to remove.